zip-0221.rst

```diff
@@ -0,0 +1,743 @@
+  check the consistency of treestate transitions over a range of blocks, without
+  recalculating the root from genesis.
+
+- ``hashLatestSaplingRoot``
+
+  * This commitment serves the same purpose as ``hashFinalSaplingRoot`` in current Sapling
+    semantics.
+  * However, because the MMR tree commits to blocks ``B_x ... B_(n-1)``, the latest
+    commitment will describe the final treestate of the previous block, rather than the
+    current block.
+  * Concretely: block 500 currently commits to the final treestate of block 500 in its
+    header. With this ZIP, block 500 will commit to all roots up to block 499, but not the
+    final root of block 500.
+  * We feel this is an acceptable tradeoff. Using the most recent treestate as a
+    transaction anchor is already unsafe in reorgs. Clients should never use the most
+    recent treestate to generate transactions, so it is acceptable to delay commitment by
+    one block.
+
+- ``nSaplingTxCount``
+
+  * By committing to the number of Sapling transactions in blocks (and ranges of blocks),
+    a light client may reliably learn whether a malicious server is witholding any
+    Sapling transactions.
+  * In addition, this commitment allows light clients to avoid syncing header ranges that
+    do not contain Sapling transactions. As the primary cost of a light client is
+    transmission of equihash solution information in block headers, this optimization
+    would significantly decrease the bandwidth requirements of light clients.
+  * An earlier draft of this ZIP committed to the number of shielded transactions,
+    counting both Sprout and Sapling. This commitment would not have been useful to light
+    clients that only support Sapling addresses; they would not be able to distinguish
+    between Sapling transactions being maliciously withheld, and Sprout transactions not
+    being requested.
+  * A commitment to the number of Sprout transactions in blocks was not included, because
+    Sprout addresses are effectively deprecated at this point, and will not be supported
+    by any light clients.
+  * If a future network upgrade introduced a new shielded pool, a new commitment to that
+    pool's transactions would be added, to similarly enable future light clients that do
+    not support Sapling addresses.
+
+Header Format Change
+--------------------
+
+The primary goal of the original authors was to minimize header changes; in particular,
+they preferred not to introduce changes that could affect mining hardware or embedded
+software. Altering the block header format would require changes throughout the ecosystem,
+so we decided against adding ``hashChainHistoryRoot`` to the header as a new field.
+
+ZIP 301 states that "[Miner client software] SHOULD alert the user upon receiving jobs
+containing block header versions they do not know about or support, and MUST ignore such
+jobs." [#zip-0301]_ As the only formally-defined block header version is 4, any header
```

Comment:
```suggestion
jobs." [#zip-0301]_ As the only formally defined block header version is 4, any header
```

("formally defined" isn't a compound adjective; "formally" is an adverb modifying "defined".)